test/core/tsi/s2a/s2a_handshaker_client_test.cc

```diff
@@ -0,0 +1,265 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/tsi/s2a/handshaker/s2a_handshaker_client.h"
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
+#include "src/core/lib/surface/channel.h"
+#include "src/core/tsi/s2a/handshaker/s2a_handshaker_util.h"
+#include "src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.h"
+#include "src/core/tsi/s2a/s2a_constants.h"
+#include "src/core/tsi/transport_security_grpc.h"
+#include "src/core/tsi/transport_security_interface.h"
+
+using ::experimental::grpc_s2a_credentials_options;
+using ::experimental::grpc_s2a_credentials_options_create;
+using ::experimental::grpc_s2a_credentials_options_destroy;
+
+namespace grpc_core {
```

Comment:
Could you please use C++ style for testing (using google test library e.g., TEST_P)? there does not seem any reason to keep writing C-style tests.

Reply:
Do you mind if I do this in a separate PR? I didn't realize I could use the test library in the C core, so I will need to change it across all of the S2A client.

Reply:
Sure. Feel free to do that.